File: backend/src/schemas/connector.ts
import { z } from "zod";

export const connectorCreateSchema = z.object({
  kind: z.string(),
  name: z.string(),
  config: z.record(z.any()).nullable().optional(),
  sync_frequency: z.string().default("1 hour"),
});

export const connectorUpdateSchema = z.object({
  name: z.string().optional(),
  config: z.record(z.any()).nullable().optional(),
  sync_frequency: z.string().optional(),
});

File: backend/app/schemas/connector.py
from pydantic import BaseModel, Field


class ConnectorCreateSchema(BaseModel):
    kind: str = "google_drive"
    name: str
    config: dict | None = None
    sync_frequency: str = "1 hour"


class ConnectorUpdateSchema(BaseModel):
    name: str | None = None
    config: dict | None = None
    sync_frequency: str | None = None